tests/regression/src/lib.rs

```diff
@@ -329,4 +331,74 @@ mod tests {
         })
         .unwrap();
     }
+
+    /// Test to measure session resumption by performing a handshake and resuming the handshake with a session ticket
+    #[test]
+    fn test_session_resumption() {
+        valgrind_test("test_session_resumption", |ctrl| {
+            const KEY_NAME: &str = "InsecureTestKey";
+            const KEY_VALUE: [u8; 16] = [3, 1, 4, 1, 5, 9, 2, 6, 5, 3, 5, 8, 9, 7, 9, 3];
+            ctrl.stop_instrumentation();
+            let keypair_rsa = CertKeyPair::default();
+
+            let mut server_config_builder = Builder::new();
+            server_config_builder
+                .add_session_ticket_key(
+                    KEY_NAME.as_bytes(),
+                    KEY_VALUE.as_slice(),
+                    SystemTime::now(),
+                )?
+                .load_pem(keypair_rsa.cert(), keypair_rsa.key())?
+                .set_security_policy(&security::DEFAULT_TLS13)?;
+            let server_config = server_config_builder.build()?;
+
+            let mut client_config_builder = Builder::new();
+            client_config_builder
+                .enable_session_tickets(true)?
+                .trust_pem(keypair_rsa.cert())?
+                .set_verify_host_callback(InsecureAcceptAllCertificatesHandler {})?
+                .set_security_policy(&security::DEFAULT_TLS13)?;
+            let client_config = client_config_builder.build()?;
+
+            // 1st handshake: no session ticket, so no resumption
+            let session_ticket = {
+                let mut pair = TestPair::from_configs(&client_config, &server_config);
+                pair.client.set_waker(Some(&noop_waker()))?;
+                pair.handshake()?;
+                assert!(pair.client.poll_recv(&mut [0]).is_pending());
+
+                let mut ticket: Vec<u8> = vec![0; pair.client.session_ticket_length().unwrap()];
+                pair.client.session_ticket(&mut ticket).unwrap();
+
+                assert!(!pair.client.resumed());
+                ticket
+            };
+
+            // Start Cachegrind instrumentation for the second handshake
+            ctrl.start_instrumentation();
```

Comment:
This is an interesting place to start measuring the performance difference. On the one hand, it does make sense to do that since we want to ensure the resumed handshake is speedy. However, I am also interested in the performance of the first handshake, since the server actually creates and sends the session ticket there. Why not start the instrumentation before the first handshake?

Reply:
Yes, I think the main consideration was that since we already test a handshake and we want these tests to be mutually exclusive, this test should cover the second handshake. Mutual exclusivity is important so we can pinpoint the areas where a regression occurs. However, if we are interested in measuring the creation of the session ticket (which is mutually exclusive form a regular handshake), we can wrap the `.session_ticket` call so it is included in instrumentation. I've included that in the latest PR.